Problem4_Cat.cpp

```diff
@@ -14,3 +16,63 @@
 // Write CppUnitLite tests to verify correct behavior.
 //
 // Which is the better approach? Explain, in a code-comment block, your rationale for which is the better approach.
+
+std::unique_ptr<char[]> cat(char* c1, char* c2)
+{
+    auto pvalues{ std::make_unique<char[]>(std::strlen(c1) + std::strlen(c2) + 1) };
+    if (c1 == nullptr || c2 == nullptr) {
+        if (c1 == nullptr && c2 != nullptr) {
+            pvalues[0] = *c2;
+        }
+        else if (c1 != nullptr && c2 == nullptr) {
+            pvalues[0] = *c1;
+        }
+        else {
+            pvalues = nullptr;
+        }
+        return pvalues;
+    }
+    /*pvalues[0] = *c1;
+    pvalues[std::strlen(c1)] = *c2;*/
+    for (size_t i{}; i < std::strlen(c1); i++) {
+        pvalues[i] = *(c1 + i);
+    }
+    for (size_t i{ std::strlen(c1) }; i < (std::strlen(c1) + std::strlen(c2)); i++) {
+        pvalues[i] = *(c2 + i- std::strlen(c1));
+    }
+    //pvalues[1] = *c2;
+
+    return pvalues;
+}
+
+TEST(concatenate, charType)
+{
+    char* a = (char*)"Hello";
+    char* b = (char*)"World";
+    auto catValue = cat(a, b);
+    std::string helloW(catValue.get());
+    //std::cout << "\n" << helloW << "\n";
+    CHECK_EQUAL("HelloWorld", catValue.get());
+
+}
+
+std::string cat2(const std::string& c1, const std::string& c2)
+{
+    std::string values;
+    values = c1 + c2;
+
+    return values;
```

Comment:
```suggestion
    return c1 + c2;
```
Correct, but why say in 3 lines what you can say in 1?